test_package/CMakeLists.txt

```diff
@@ -4,7 +4,7 @@ project(test_dummy
 	LANGUAGES NONE
 )
 
-find_package(dummy REQUIRED)
+find_package(dummy REQUIRED CONFIG)
 message("dummy_MODULE_PATH: ${dummy_MODULE_PATH}")
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${dummy_MODULE_PATH})
```

Comment:
I do not think you need an extra variable ```dummy_MODULE_PATH``` in your ```dummy-config.cmake``` to get the dummy-filepath as ```find_package``` already gives you this information by default with ```dummy_DIR``` ([see here](https://cmake.org/cmake/help/v3.14/command/find_package.html?highlight=%3CPackageName%3E_DIR#full-signature-and-config-mode)).

Regardless what your ```dummy-config.cmake``` contains (even if it is empty) with this approach the path to the ```dummy-config.cmake``` will be appended to ```CMAKE_MODULE_PATH``` correctly.

```cmake
find_package (dummy REQUIRED CONFIG)
set (CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${dummy_DIR}")
```

Reply:
Just to add to clarify ```find_package``` relies on [CMAKE_MODULE_PATH](https://cmake.org/cmake/help/v3.14/variable/CMAKE_MODULE_PATH.html#variable:CMAKE_MODULE_PATH) in the [basic mode](https://cmake.org/cmake/help/v3.14/command/find_package.html?highlight=NAMES#basic-signature-and-module-mode) and on [CMAKE_PREFIX_PATH](CMAKE_PREFIX_PATH) in [full/config mode](https://cmake.org/cmake/help/v3.14/command/find_package.html#full-signature-and-config-mode) according to the CMake documentation.

Reply:
About your first point!

First you're right and based on my repository that would be definitively the best option!

In my real applications in `${dummy_DIR}`, there are 2 folders:
- `modules` which is kind of a cmake library with many cmake files (including macros, functions, etc...) 
- `find_modules` which aggregates basic `Find*.cmake` we used in our projects.

In practice there are 2 variables, that we use:
- `dummy_MODULES_DIR` = `${dummy_DIR}/modules`
- `dummy_FIND_MODULES_DIR` = `${dummy_DIR}/find_modules`

Finally, we define `dummy_MODULE_PATH` as follows:
```cmake
set(dummy_MODULE_PATH ${dummy_DIR}/modules ${dummy_DIR}/find_modules)
```

With your proposal, I would have to change:
```cmake
set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${dummy_MODULE_PATH})
```
to
```cmake
set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${dummy_DIR}/modules ${dummy_DIR}/find_modules)
```
It would also work, but to my opinion it was easier for consumer to rely on the previous one.

Reply:
About your second point, I think this is an excellent point to be considered especially in `test_package/conanfile.py`. Setting both `CMAKE_PREFIX_PATH`and `CMAKE_MODULE_PATH` would make the approach even more robust.

What do you think?

Reply:
first point: Yes, if your dummy package does provide further paths I would set them in the ```dummy-config.cmake``` as external files should know as little as possible about anything in the dummy package. This way depending ```CMakeLists.txt``` files only need to ```find_package``` in the right order (first dummy, then "sub-dummy") without path fiddeling.

dummy-config.cmake (do not forget quotes for paths strings):
```cmake
set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_CURRENT_LIST_DIR}/modules" "${CMAKE_CURRENT_LIST_DIR}/find_modules")
```
CMakeLists.cmake:
```cmake
find_package (dummy ...)
find_package (sub-dummy ...)
```


second point: Setting both ```CMAKE_PREFIX_PATH``` and ```CMAKE_MODULE_PATH``` will not hurt at the moment. But I would advise you to specify as little as possible to minimize side-effects in the future. Unsetting configurations is likely to bring problems to existing code bases but extending configurations does not hurt normally.
Because if you use for example only FindXXX.cmake files you do not need ```CMAKE_PREFIX_PATH``` but setting it yould influence ```find_program```, ```find_library```, etc. what maybe was not desired.
In your case you have to set the ```CMAKE_PREFIX_PATH``` as you use the full config mode. If you use FindXXX.cmake files (basic/module mode) too, then you have to set both.

Reply:
@Johnnyxy Thank you so much for all your valuable comments!!
I'll keep this repository and the PR opened for reference